src/mime/mod.rs

```diff
@@ -109,6 +109,33 @@ impl Mime {
             })
             .flatten()
     }
+
+    /// Remove a param from the set. Returns the `ParamValue` if it was contained within the set.
+    pub fn remove_param(&mut self, name: impl Into<ParamName>) -> Option<ParamValue> {
```

Comment:
Should this be `take_param`? (Idk.)

Reply:
This is modeled after [`HashMap::remove`](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.remove) -- it removes the value; however if the entry existed it returns it.